File: android/app/src/main/java/com/lichso/app/feature/content/KnowledgeFeedScreen.kt
package com.lichso.app.feature.content

import com.lichso.app.ui.theme.screenBackground
import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.Article
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.material3.pulltorefresh.PullToRefreshBox
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import coil.compose.AsyncImage
import androidx.compose.ui.layout.ContentScale
import com.lichso.app.data.remote.*
import com.lichso.app.ui.components.AppTopBar
import com.lichso.app.ui.components.LichSoSkeletonList
import com.lichso.app.ui.theme.LichSoThemeColors
import kotlinx.coroutines.flow.distinctUntilChanged
import java.time.LocalDate
import java.time.format.DateTimeFormatter

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun KnowledgeFeedScreen(
    onBackClick: () -> Unit = {},
    onAskAi: (String) -> Unit = {},
    onArticleClick: (String) -> Unit = {},
    viewModel: KnowledgeFeedViewModel = hiltViewModel(),
) {
    val c = LichSoThemeColors.current
    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
    val listState = rememberLazyListState()

    LaunchedEffect(listState, uiState.articles.size, uiState.hasMoreArticles, uiState.isLoadingMore) {
        snapshotFlow {
            val layoutInfo = listState.layoutInfo
            val lastVisibleIndex = layoutInfo.visibleItemsInfo.lastOrNull()?.index ?: 0
            lastVisibleIndex >= layoutInfo.totalItemsCount - 4
        }
            .distinctUntilChanged()
            .collect { shouldLoadMore ->
                if (shouldLoadMore) viewModel.loadMoreArticles()
            }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .screenBackground(c.bg)
    ) {
        AppTopBar(
            title = "Khám Phá",
            subtitle = "Kiến thức hôm nay",
            onBackClick = onBackClick,
            actions = {
                IconButton(onClick = { viewModel.refresh() }) {
                    Icon(
                        Icons.Filled.Refresh,
                        contentDescription = "Làm mới",
                        tint = Color.White,
                    )
                }
            }
        )

        if (uiState.isLoading) {
            LichSoSkeletonList(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(top = 12.dp),
                itemCount = 6,
            )
        } else {
            PullToRefreshBox(
                isRefreshing = uiState.isRefreshing,
                onRefresh = { viewModel.refresh() },
                modifier = Modifier.fillMaxSize(),
            ) {
            LazyColumn(
                modifier = Modifier.fillMaxSize(),
                state = listState,
                contentPadding = PaddingValues(bottom = 24.dp),
            ) {
                uiState.quote?.let { quote ->
                    item {
                        QuoteCard(quote = quote, onAskAi = onAskAi)
                    }
                }

                if (uiState.events.isNotEmpty()) {
                    item {
                        SectionHeader(title = "Ngày này năm xưa")
                    }
                    item {
                        LazyRow(
                            contentPadding = PaddingValues(horizontal = 16.dp),
                            horizontalArrangement = Arrangement.spacedBy(12.dp),
                        ) {
                            items(uiState.events) { event ->
                                EventCard(event = event)
                            }
                        }
                    }
                }

                if (uiState.famousPeople.isNotEmpty()) {
                    item {
                        SectionHeader(title = "Nhân vật sinh nhật hôm nay")
                    }
                    item {
                        LazyRow(
                            contentPadding = PaddingValues(horizontal = 16.dp),
                            horizontalArrangement = Arrangement.spacedBy(12.dp),
                        ) {
                            items(uiState.famousPeople) { person ->
                                PersonCard(person = person)
                            }
                        }
                    }
                }

                if (uiState.categories.isNotEmpty()) {
                    item {
                        SectionHeader(title = "Danh mục bài viết")
                    }
                    item {
                        LazyRow(
                            contentPadding = PaddingValues(horizontal = 16.dp),
                            horizontalArrangement = Arrangement.spacedBy(12.dp),
                        ) {
                            item {
                                CategoryCard(
                                    name = "Tất cả",
                                    imageUrl = null,
                                    selected = uiState.selectedCategorySlug == null,
                                    onClick = { viewModel.selectCategory(null) },
                                )
                            }
                            items(uiState.categories) { category ->
                                CategoryCard(
                                    name = category.name,
                                    imageUrl = category.imageUrl,
                                    selected = uiState.selectedCategorySlug == category.slug,
                                    onClick = { viewModel.selectCategory(category) },
                                )
                            }
                        }
                    }
                }

                if (uiState.articles.isNotEmpty()) {
                    item {
                        SectionHeader(title = uiState.selectedCategoryName ?: "Bài viết mới nhất")
                    }
                    items(uiState.articles) { article ->
                        ArticleCard(article = article, onClick = { onArticleClick(article.id) })
                    }
                    if (uiState.isLoadingMore) {
                        item {
                            Box(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(vertical = 16.dp),
                                contentAlignment = Alignment.Center,
                            ) {
                                CircularProgressIndicator(
                                    color = c.red,
                                    modifier = Modifier.size(24.dp),
                                    strokeWidth = 2.dp,
                                )
                            }
                        }
                    } else if (!uiState.hasMoreArticles && uiState.articles.isNotEmpty()) {
                        item {
                            Text(
                                text = "Bạn đã xem hết bài viết hiện có",
                                style = TextStyle(fontSize = 12.sp, color = c.textTertiary),
                                textAlign = TextAlign.Center,
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(vertical = 16.dp),
                            )
                        }
                    }
                } else if (uiState.categories.isNotEmpty() && uiState.isLoadingMore) {
                    item {
                        Box(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 24.dp),
                            contentAlignment = Alignment.Center,
                        ) {
                            CircularProgressIndicator(
                                color = c.red,
                                modifier = Modifier.size(24.dp),
                                strokeWidth = 2.dp,
                            )
                        }
                    }
                } else if (uiState.categories.isNotEmpty() && uiState.selectedCategorySlug != null) {
                    item {
                        Text(
                            text = "Chưa có bài viết trong danh mục này",
                            style = TextStyle(fontSize = 13.sp, color = c.textTertiary),
                            textAlign = TextAlign.Center,
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 16.dp, vertical = 24.dp),
                        )
                    }
                }

                if (uiState.events.isEmpty() && uiState.famousPeople.isEmpty() && uiState.articles.isEmpty() && uiState.quote == null) {
                    item {
                        Box(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(top = 64.dp),
                            contentAlignment = Alignment.Center,
                        ) {
                            Text(
                                "Không có dữ liệu hôm nay",
                                style = TextStyle(fontSize = 14.sp, color = c.textTertiary),
                            )
                        }
                    }
                }
            }
            }
        }
    }
}

@Composable
private fun CategoryCard(
    name: String,
    imageUrl: String?,
    selected: Boolean,
    onClick: () -> Unit,
) {
    val c = LichSoThemeColors.current
    val normalizedUrl = remember(imageUrl) { normalizeContentImageUrl(imageUrl) }

    Box(
        modifier = Modifier
            .width(132.dp)
            .height(92.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(c.surfaceContainer)
            .border(
                width = if (selected) 2.dp else 1.dp,
                color = if (selected) c.gold else c.outlineVariant,
                shape = RoundedCornerShape(16.dp),
            )
            .clickable(onClick = onClick)
    ) {
        if (!normalizedUrl.isNullOrBlank()) {
            AsyncImage(
                model = normalizedUrl,
                contentDescription = null,
                contentScale = ContentScale.Crop,
                modifier = Modifier.matchParentSize(),
            )
            Box(
                modifier = Modifier
                    .matchParentSize()
                    .background(
                        Brush.verticalGradient(
                            colors = listOf(Color.Transparent, Color.Black.copy(alpha = 0.72f)),
                        )
                    )
            )
        } else {
            Box(
                modifier = Modifier
                    .matchParentSize()
                    .background(
                        Brush.linearGradient(
                            colors = listOf(c.red.copy(alpha = 0.82f), c.gold.copy(alpha = 0.72f)),
                            start = Offset(0f, 0f),
                            end = Offset(Float.POSITIVE_INFINITY, Float.POSITIVE_INFINITY),
                        )
                    )
            )
            Icon(
                Icons.AutoMirrored.Filled.Article,
                contentDescription = null,
                tint = Color.White.copy(alpha = 0.36f),
                modifier = Modifier
                    .align(Alignment.TopEnd)
                    .padding(10.dp)
                    .size(30.dp),
            )
        }
        Text(
            text = name,
            style = TextStyle(
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
                lineHeight = 17.sp,
            ),
            maxLines = 2,
            overflow = TextOverflow.Ellipsis,
            modifier = Modifier
                .align(Alignment.BottomStart)
                .padding(12.dp),
        )
    }
}

@Composable
private fun QuoteCard(quote: Quote, onAskAi: (String) -> Unit) {
    val c = LichSoThemeColors.current
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 12.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(
                Brush.linearGradient(
                    colors = listOf(Color(0xFF8B0000), Color(0xFFD4A017)),
                    start = Offset(0f, 0f),
                    end = Offset(Float.POSITIVE_INFINITY, Float.POSITIVE_INFINITY),
                )
            )
            .clickable { onAskAi("Giải thích sâu hơn về câu nói: \"${quote.quote}\"") }
            .padding(16.dp)
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(6.dp),
            ) {
                Icon(
                    Icons.Filled.FormatQuote,
                    contentDescription = null,
                    tint = Color.White.copy(alpha = 0.7f),
                    modifier = Modifier.size(18.dp),
                )
                Text(
                    "Danh ngôn hôm nay",
                    style = TextStyle(
                        fontSize = 11.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color.White.copy(alpha = 0.7f),
                    ),
                )
            }
            Text(
                quote.quote,
                style = TextStyle(
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color.White,
                    lineHeight = 22.sp,
                ),
            )
            if (!quote.author.isNullOrBlank()) {
                Text(
                    "— ${quote.author}",
                    style = TextStyle(
                        fontSize = 12.sp,
                        color = Color.White.copy(alpha = 0.75f),
                    ),
                )
            }
        }
    }
}

@Composable
private fun SectionHeader(title: String) {
    val c = LichSoThemeColors.current
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Box(
            modifier = Modifier
                .width(3.dp)
                .height(18.dp)
                .background(c.red, RoundedCornerShape(2.dp))
        )
        Text(
            title,
            style = TextStyle(
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = c.textPrimary,
            ),
        )
    }
}

@Composable
private fun EventCard(event: ContentEvent) {
    val c = LichSoThemeColors.current
    Box(
        modifier = Modifier
            .width(200.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(c.surfaceContainer)
            .border(1.dp, c.outlineVariant, RoundedCornerShape(12.dp))
            .padding(12.dp)
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
            if (event.eventYear != null) {
                Box(
                    modifier = Modifier
                        .clip(RoundedCornerShape(6.dp))
                        .background(c.red.copy(alpha = 0.15f))
                        .padding(horizontal = 8.dp, vertical = 3.dp)
                ) {
                    Text(
                        "${event.eventYear}",
                        style = TextStyle(
                            fontSize = 12.sp,
                            fontWeight = FontWeight.Bold,
                            color = c.red,
                        ),
                    )
                }
            }
            Text(
                event.title,
                style = TextStyle(
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = c.textPrimary,
                    lineHeight = 18.sp,
                ),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            val desc = event.shortDescription ?: event.description
            if (!desc.isNullOrBlank()) {
                Text(
                    desc,
                    style = TextStyle(
                        fontSize = 11.sp,
                        color = c.textSecondary,
                        lineHeight = 16.sp,
                    ),
                    maxLines = 3,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }
    }
}

@Composable
private fun PersonCard(person: FamousPerson) {
    val c = LichSoThemeColors.current
    val initial = person.name.firstOrNull()?.uppercase() ?: "?"
    Box(
        modifier = Modifier
            .width(140.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(c.surfaceContainer)
            .border(1.dp, c.outlineVariant, RoundedCornerShape(12.dp))
            .padding(12.dp)
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(CircleShape)
                    .background(
                        Brush.linearGradient(
                            colors = listOf(Color(0xFF1565C0), Color(0xFF7B1FA2)),
                            start = Offset(0f, 0f),
                            end = Offset(Float.POSITIVE_INFINITY, Float.POSITIVE_INFINITY),
                        )
                    ),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    initial,
                    style = TextStyle(
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.White,
                    ),
                )
            }
            Text(
                person.name,
                style = TextStyle(
                    fontSize = 12.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = c.textPrimary,
                    lineHeight = 16.sp,
                ),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            if (person.birthYear != null) {
                Text(
                    "Sinh ${person.birthYear}",
                    style = TextStyle(
                        fontSize = 11.sp,
                        color = c.textTertiary,
                    ),
                )
            }
        }
    }
}

@Composable
private fun ArticleCard(article: Article, onClick: () -> Unit = {}) {
    val c = LichSoThemeColors.current
    val featuredImage = article.featuredImage
    val normalizedUrl = remember(featuredImage) { normalizeContentImageUrl(featuredImage) }

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 6.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(c.surfaceContainer)
            .clickable(onClick = onClick)
            .border(1.dp, c.outlineVariant, RoundedCornerShape(12.dp))
            .padding(12.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalAlignment = Alignment.Top,
    ) {
        if (!normalizedUrl.isNullOrBlank()) {
            AsyncImage(
                model = normalizedUrl,
                contentDescription = null,
                contentScale = ContentScale.Crop,
                modifier = Modifier
                    .size(64.dp)
                    .clip(RoundedCornerShape(8.dp))
                    .background(c.bg3)
            )
        } else {
            Box(
                modifier = Modifier
                    .size(64.dp)
                    .clip(RoundedCornerShape(8.dp))
                    .background(c.bg3),
                contentAlignment = Alignment.Center,
            ) {
                Icon(
                    Icons.AutoMirrored.Filled.Article,
                    contentDescription = null,
                    tint = c.textTertiary,
                    modifier = Modifier.size(28.dp),
                )
            }
        }
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Text(
                article.title,
                style = TextStyle(
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = c.textPrimary,
                    lineHeight = 18.sp,
                ),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            if (!article.category?.name.isNullOrBlank()) {
                Box(
                    modifier = Modifier
                        .clip(RoundedCornerShape(4.dp))
                        .background(c.gold.copy(alpha = 0.15f))
                        .padding(horizontal = 6.dp, vertical = 2.dp)
                ) {
                    Text(
                        article.category?.name.orEmpty(),
                        style = TextStyle(
                            fontSize = 10.sp,
                            fontWeight = FontWeight.Medium,
                            color = c.gold,
                        ),
                    )
                }
            }
            if (!article.publishedAt.isNullOrBlank()) {
                Text(
                    article.publishedAt.take(10),
                    style = TextStyle(
                        fontSize = 11.sp,
                        color = c.textTertiary,
                    ),
                )
            }
        }
    }
}

private fun normalizeContentImageUrl(url: String?): String? {
    if (url.isNullOrBlank()) return null
    if (url.startsWith("http://") || url.startsWith("https://")) return url

    val cleaned = if (url.startsWith("/")) url.substring(1) else url
    return when {
        cleaned.startsWith("api/uploads/") -> "https://lichso.vn/$cleaned"
        cleaned.startsWith("uploads/") -> "https://lichso.vn/api/$cleaned"
        else -> "https://lichso.vn/$cleaned"
    }
}

File: ios/LichSo/LichSo/Screen/Tools/KnowledgeFeedScreen.swift
import SwiftUI

// ═══════════════════════════════════════════
// KnowledgeFeedScreen — màn "Khám Phá" (port giống Android KnowledgeFeedScreen.kt).
// Danh ngôn hôm nay · Ngày này năm xưa · Nhân vật sinh nhật · Danh mục · Bài viết
// (phân trang + kéo làm mới + cuộn vô hạn). Nguồn dữ liệu: server (không còn dùng
// HistoricalEventProvider cục bộ như bản iOS cũ).
// ═══════════════════════════════════════════

struct KnowledgeFeedScreen: View {
    @Environment(\.dismiss) private var dismiss
    @StateObject private var vm = KnowledgeFeedViewModel()

    @State private var aiPrompt = ""
    @State private var showAi = false

    var body: some View {
        VStack(spacing: 0) {
            topBar

            if vm.isLoading {
                loadingSkeleton
            } else {
                ScrollView {
                    LazyVStack(alignment: .leading, spacing: 0) {
                        if let quote = vm.quote {
                            QuoteCard(quote: quote) {
                                aiPrompt = "Giải thích sâu hơn về câu nói: \"\(quote.quote)\""
                                showAi = true
                            }
                        }

                        if !vm.events.isEmpty {
                            sectionHeader("Ngày này năm xưa")
                            ScrollView(.horizontal, showsIndicators: false) {
                                HStack(spacing: 12) {
                                    ForEach(vm.events) { EventCard(event: $0) }
                                }
                                .padding(.horizontal, 16)
                            }
                        }

                        if !vm.famousPeople.isEmpty {
                            sectionHeader("Nhân vật sinh nhật hôm nay")
                            ScrollView(.horizontal, showsIndicators: false) {
                                HStack(spacing: 12) {
                                    ForEach(vm.famousPeople) { PersonCard(person: $0) }
                                }
                                .padding(.horizontal, 16)
                            }
                        }

                        if !vm.categories.isEmpty {
                            sectionHeader("Danh mục bài viết")
                            ScrollView(.horizontal, showsIndicators: false) {
                                HStack(spacing: 12) {
                                    CategoryCard(name: "Tất cả", imageUrl: nil,
                                                 selected: vm.selectedCategorySlug == nil) {
                                        vm.selectCategory(nil)
                                    }
                                    ForEach(vm.categories) { category in
                                        CategoryCard(name: category.name, imageUrl: category.imageUrl,
                                                     selected: vm.selectedCategorySlug == category.slug) {
                                            vm.selectCategory(category)
                                        }
                                    }
                                }
                                .padding(.horizontal, 16)
                            }
                        }

                        articlesSection
                        emptyStateIfNeeded
                    }
                    .padding(.bottom, 24)
                }
                .refreshable { await vm.load(isRefresh: true) }
            }
        }
        .background(LSTheme.bg.ignoresSafeArea())
        .navigationBarHidden(true)
        .onAppear { vm.loadIfNeeded() }
        .fullScreenCover(isPresented: $showAi) {
            NavigationStack { AIChatScreen(initialMessage: aiPrompt) }
        }
    }

    // ── Top bar (đỏ gradient giống AppTopBar Android) ──
    private var topBar: some View {
        HStack(spacing: 12) {
            Button { dismiss() } label: {
                Image(systemName: "chevron.left")
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(width: 40, height: 40)
            }
            VStack(alignment: .leading, spacing: 2) {
                Text("Khám Phá")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(.white)
                Text("Kiến thức hôm nay")
                    .font(.system(size: 12))
                    .foregroundColor(.white.opacity(0.85))
            }
            Spacer()
            Button { vm.refresh() } label: {
                Image(systemName: "arrow.clockwise")
                    .font(.system(size: 17, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(width: 40, height: 40)
            }
        }
        .padding(.horizontal, 8)
        .padding(.bottom, 10)
        .background(
            LinearGradient(
                colors: [Color(red: 0.773, green: 0.157, blue: 0.157),
                         Color(red: 0.545, green: 0, blue: 0)],
                startPoint: .topLeading, endPoint: .bottomTrailing
            )
            .ignoresSafeArea(edges: .top)
        )
    }

    private var loadingSkeleton: some View {
        VStack(spacing: 12) {
            ForEach(0..<6, id: \.self) { _ in
                RoundedRectangle(cornerRadius: 12)
                    .fill(LSTheme.surfaceContainer)
                    .frame(height: 84)
                    .padding(.horizontal, 16)
            }
            Spacer()
        }
        .padding(.top, 12)
        .redacted(reason: .placeholder)
    }

    @ViewBuilder private var articlesSection: some View {
        if !vm.articles.isEmpty {
            sectionHeader(vm.selectedCategoryName ?? "Bài viết mới nhất")
            ForEach(vm.articles) { article in
                NavigationLink(destination: ArticleDetailScreen(articleId: article.id)) {
                    ArticleCard(article: article)
                }
                .buttonStyle(.plain)
                .onAppear {
                    if article.id == vm.articles.last?.id { vm.loadMoreArticles() }
                }
            }
            if vm.isLoadingMore {
                centerSpinner.padding(.vertical, 16)
            } else if !vm.hasMoreArticles {
                Text("Bạn đã xem hết bài viết hiện có")
                    .font(.system(size: 12))
                    .foregroundColor(LSTheme.textTertiary)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 16)
            }
        } else if !vm.categories.isEmpty && vm.isLoadingMore {
            centerSpinner.padding(.vertical, 24)
        } else if !vm.categories.isEmpty && vm.selectedCategorySlug != nil {
            Text("Chưa có bài viết trong danh mục này")
                .font(.system(size: 13))
                .foregroundColor(LSTheme.textTertiary)
                .frame(maxWidth: .infinity)
                .padding(.horizontal, 16).padding(.vertical, 24)
        }
    }

    @ViewBuilder private var emptyStateIfNeeded: some View {
        if vm.events.isEmpty && vm.famousPeople.isEmpty && vm.articles.isEmpty && vm.quote == nil {
            Text("Không có dữ liệu hôm nay")
                .font(.system(size: 14))
                .foregroundColor(LSTheme.textTertiary)
                .frame(maxWidth: .infinity)
                .padding(.top, 64)
        }
    }

    private var centerSpinner: some View {
        ProgressView()
            .tint(LSTheme.primary)
            .frame(maxWidth: .infinity)
    }

    private func sectionHeader(_ title: String) -> some View {
        HStack(spacing: 8) {
            RoundedRectangle(cornerRadius: 2)
                .fill(LSTheme.primary)
                .frame(width: 3, height: 18)
            Text(title)
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(LSTheme.textPrimary)
        }
        .padding(.horizontal, 16).padding(.vertical, 8)
    }
}

// ══════════════════════════════════════════
// CARDS
// ══════════════════════════════════════════

private struct QuoteCard: View {
    let quote: Quote
    let onAskAi: () -> Void

    var body: some View {
        Button(action: onAskAi) {
            VStack(alignment: .leading, spacing: 8) {
                HStack(spacing: 6) {
                    Image(systemName: "quote.opening")
                        .font(.system(size: 16))
                        .foregroundColor(.white.opacity(0.7))
                    Text("Danh ngôn hôm nay")
                        .font(.system(size: 11, weight: .semibold))
                        .foregroundColor(.white.opacity(0.7))
                }
                Text(quote.quote)
                    .font(.system(size: 15, weight: .medium))
                    .foregroundColor(.white)
                    .lineSpacing(5)
                    .frame(maxWidth: .infinity, alignment: .leading)
                if let author = quote.author, !author.isEmpty {
                    Text("— \(author)")
                        .font(.system(size: 12))
                        .foregroundColor(.white.opacity(0.75))
                }
            }
            .padding(16)
            .background(
                LinearGradient(
                    colors: [Color(hex: "8B0000"), Color(hex: "D4A017")],
                    startPoint: .topLeading, endPoint: .bottomTrailing
                )
            )
            .clipShape(RoundedRectangle(cornerRadius: 16))
        }
        .buttonStyle(.plain)
        .padding(.horizontal, 16).padding(.vertical, 12)
    }
}

private struct EventCard: View {
    let event: ContentEvent
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            if let year = event.eventYear {
                Text("\(year)")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundColor(LSTheme.primary)
                    .padding(.horizontal, 8).padding(.vertical, 3)
                    .background(LSTheme.primary.opacity(0.15))
                    .clipShape(RoundedRectangle(cornerRadius: 6))
            }
            Text(event.title)
                .font(.system(size: 13, weight: .semibold))
                .foregroundColor(LSTheme.textPrimary)
                .lineLimit(2)
            let desc = event.shortDescription ?? event.description
            if let desc, !desc.isEmpty {
                Text(desc)
                    .font(.system(size: 11))
                    .foregroundColor(LSTheme.textSecondary)
                    .lineLimit(3)
            }
        }
        .padding(12)
        .frame(width: 200, alignment: .leading)
        .background(LSTheme.surfaceContainer)
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(LSTheme.outlineVariant, lineWidth: 1))
    }
}

private struct PersonCard: View {
    let person: FamousPerson
    private var initial: String { String(person.name.first ?? "?").uppercased() }
    var body: some View {
        VStack(spacing: 8) {
            ZStack {
                Circle()
                    .fill(LinearGradient(
                        colors: [Color(hex: "1565C0"), Color(hex: "7B1FA2")],
                        startPoint: .topLeading, endPoint: .bottomTrailing
                    ))
                    .frame(width: 48, height: 48)
                Text(initial)
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(.white)
            }
            Text(person.name)
                .font(.system(size: 12, weight: .semibold))
                .foregroundColor(LSTheme.textPrimary)
                .multilineTextAlignment(.center)
                .lineLimit(2)
            if let year = person.birthYear {
                Text("Sinh \(year)")
                    .font(.system(size: 11))
                    .foregroundColor(LSTheme.textTertiary)
            }
        }
        .padding(12)
        .frame(width: 140)
        .background(LSTheme.surfaceContainer)
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(LSTheme.outlineVariant, lineWidth: 1))
    }
}

private struct CategoryCard: View {
    let name: String
    let imageUrl: String?
    let selected: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            ZStack(alignment: .bottomLeading) {
                if let normalized = normalizeContentImageUrl(imageUrl), let url = URL(string: normalized) {
                    CachedAsyncImage(url: url) { image in
                        image.resizable().aspectRatio(contentMode: .fill)
                    } placeholder: {
                        LSTheme.surfaceContainer
                    }
                    .frame(width: 132, height: 92)
                    .clipped()
                    LinearGradient(colors: [.clear, .black.opacity(0.72)],
                                   startPoint: .top, endPoint: .bottom)
                } else {
                    LinearGradient(
                        colors: [LSTheme.primary.opacity(0.82), LSTheme.gold.opacity(0.72)],
                        startPoint: .topLeading, endPoint: .bottomTrailing
                    )
                    Image(systemName: "doc.text")
                        .font(.system(size: 26))
                        .foregroundColor(.white.opacity(0.36))
                        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topTrailing)
                        .padding(10)
                }
                Text(name)
                    .font(.system(size: 13, weight: .bold))
                    .foregroundColor(.white)
                    .lineLimit(2)
                    .padding(12)
            }
            .frame(width: 132, height: 92)
            .clipShape(RoundedRectangle(cornerRadius: 16))
            .overlay(
                RoundedRectangle(cornerRadius: 16)
                    .stroke(selected ? LSTheme.gold : LSTheme.outlineVariant,
                            lineWidth: selected ? 2 : 1)
            )
        }
        .buttonStyle(.plain)
    }
}

private struct ArticleCard: View {
    let article: Article
    var body: some View {
        HStack(alignment: .top, spacing: 12) {
            if let normalized = normalizeContentImageUrl(article.featuredImage), let url = URL(string: normalized) {
                CachedAsyncImage(url: url) { image in
                    image.resizable().aspectRatio(contentMode: .fill)
                } placeholder: {
                    LSTheme.surfaceContainerHigh
                }
                .frame(width: 64, height: 64)
                .clipShape(RoundedRectangle(cornerRadius: 8))
            } else {
                ZStack {
                    LSTheme.surfaceContainerHigh
                    Image(systemName: "doc.text")
                        .font(.system(size: 26))
                        .foregroundColor(LSTheme.textTertiary)
                }
                .frame(width: 64, height: 64)
                .clipShape(RoundedRectangle(cornerRadius: 8))
            }
            VStack(alignment: .leading, spacing: 4) {
                Text(article.title)
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundColor(LSTheme.textPrimary)
                    .lineLimit(2)
                if let cat = article.category?.name, !cat.isEmpty {
                    Text(cat)
                        .font(.system(size: 10, weight: .medium))
                        .foregroundColor(LSTheme.gold)
                        .padding(.horizontal, 6).padding(.vertical, 2)
                        .background(LSTheme.gold.opacity(0.15))
                        .clipShape(RoundedRectangle(cornerRadius: 4))
                }
                if let published = article.publishedAt, !published.isEmpty {
                    Text(String(published.prefix(10)))
                        .font(.system(size: 11))
                        .foregroundColor(LSTheme.textTertiary)
                }
            }
            Spacer(minLength: 0)
        }
        .padding(12)
        .background(LSTheme.surfaceContainer)
        .clipShape(RoundedRectangle(cornerRadius: 12))
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(LSTheme.outlineVariant, lineWidth: 1))
        .padding(.horizontal, 16).padding(.vertical, 6)
    }
}

// ── Chuẩn hoá URL ảnh nội dung (mirror Android normalizeContentImageUrl) ──
func normalizeContentImageUrl(_ url: String?) -> String? {
    guard let url, !url.isEmpty else { return nil }
    if url.hasPrefix("http://") || url.hasPrefix("https://") { return url }
    let cleaned = url.hasPrefix("/") ? String(url.dropFirst()) : url
    if cleaned.hasPrefix("api/uploads/") { return "https://lichso.vn/\(cleaned)" }
    if cleaned.hasPrefix("uploads/") { return "https://lichso.vn/api/\(cleaned)" }
    return "https://lichso.vn/\(cleaned)"
}
